eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -338,10 +330,50 @@ class Router(nodeParams: NodeParams, watcher: ActorRef) extends FSM[State, Data]
       val (format, theirShortChannelIds, useGzip) = ChannelRangeQueries.decodeShortChannelIds(data)
       val ourShortChannelIds: SortedSet[ShortChannelId] = d.channels.keySet.filter(keep(firstBlockNum, numberOfBlocks, _, d.channels, d.updates))
       val missing: SortedSet[ShortChannelId] = theirShortChannelIds -- ourShortChannelIds
-      log.info("{} sent their reply fr, we're missing {} channel announcements/updates, format={} useGzip={}", remoteNodeId, missing.size, format, useGzip)
+      log.info("we received their reply, we're missing {} channel announcements/updates, format={} useGzip={}", missing.size, format, useGzip)
       val blocks = ChannelRangeQueries.encodeShortChannelIds(firstBlockNum, numberOfBlocks, missing, format, useGzip)
       blocks.foreach(block => sender ! QueryShortChannelIds(chainHash, block.shortChannelIds))
-      stay
+
+      // we have channel announcement that they don't have: check if we can prune them
+      val pruningCandidates = {
```

Comment:
maybe we should just do:

```scala
val missing: SortedSet[ShortChannelId] = theirShortChannelIds -- ourShortChannelIds
...
val maybeStale: SortedSet[ShortChannelId] = ourShortChannelIds -- theirShortChannelIds
```

instead of trying to be smart?